### ISSUE TITLE
Scala 2.12 raises an `InvalidObjectException` on Java 17

### ISSUE THREAD

**Issue by skylee03** [clojure-finance/datajure#18]
```
java.io.InvalidObjectException: ReflectiveOperationException during deserialization
```

Can be solved by upgrading Scala to 2.13, but doing so will cause another error:

```
Execution error (ClassNotFoundException) at java.net.URLClassLoader/findClass (URLClassLoader.java:445).
scala.collection.convert.Wrappers$IteratorWrapper
```

In [`geni/arrow.clj`](https://github.com/zero-one-group/geni/blob/83b24ea5e87595a66766118444b7a4e1dfa1ec8c/src/clojure/zero_one/geni/arrow.clj), `(scala.collection.convert Wrappers$IteratorWrapper)` is imported, but it is removed in Scala 2.13.

**Comment by skylee03** [comment]
Newer versions of Scala 2.12 solved this problem.

**Comment by skylee03** [comment]
Solved by upgrading Spark version to 3.3.3, whose default Scala target is 2.12.15.

**Comment by skylee03** [comment]
Now can be fixed by upgrading Geni to v0.0.41 (to be released soon).
- https://github.com/zero-one-group/geni/pull/346
